### DISCUSSION TITLE
[Feature] Live Activities Widget for Foreground Back Up

### DISCUSSION THREAD

**Discussion by xpire** [immich-app/immich#8914] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

I want to see the foreground upload implemented with [Live Activities widget](https://developer.apple.com/design/human-interface-guidelines/live-activities), so I can do other things while the foreground backup occurs (e.g. lock phone, move to different app)

#### Why
I mainly use foreground back up to keep my Immich library up to date. A small issue is the back up is a bit finicky, and if I move from the app to another app or leave it running but the phone's screen turns off, the back up will usually fail. I then have to turn on the phone and keep it awake, either by tapping it every so often or changing the sleep settings to keep it on.

#### Impact
This feature would greatly improve the robustness of the backup feature, as I wouldn't have to worry whether the upload is failing when the screen turns off or I've moved to another app as it would show in the dynamic island or would be persisted in the notification centre.

Using an up-to-date feature like live activities will also be something that can entice iOS users to use immich, as not many apps take advantage of this feature. I'm also a dev and would be keen to work on contributing this feature.

### Platform

- [ ] Server
- [ ] Web
- [X] Mobile

**Comment by davidvhammond** [comment]
This would be a really nice improvement.

**Comment by radh21301** [comment]
I think this would be a great addition! I wonder if this would improve background backup as well. 

Would a scenario like this work:?
* The IOS OS lets immich get an opportunity to do a background sync. (Note: From a lot of the discussions, it seems like this event can be active for only a few seconds. And this duration is not enough to back up a lot of assests. Even worse of the internet connection is poor)
* The immich app then triggers a live activity. This would then keep the app active for however long immich wants? I would expect this to be much longer than the few seconds which we currently get.
* Once immich is done with the uploads. It can stop the live activity widget.

Seems pretty cool! Just not sure if I understand how live activity works in the first place.

**Comment by orsoFra** [comment]
I think the live activity notifications for the foreground/background backup would be a great QoL improvement for the Immich iOS app. Right now, Immich just pushes out a notification every time a photo is uploaded, which is a bit annoying.
In my mind the design could be somewhat similarlar to what Instagram uses for the story upload live notification: [(Ref)](https://9to5mac.com/2023/07/07/instagram-live-activities-ios-app/)
Plus Apple has good documentation on how to design it: [Ref](https://developer.apple.com/design/human-interface-guidelines/live-activities)

**Comment by Firedorange717** [comment]
I was just thinking the same thing, the notifications for backups are kind of weird. I feel like live activities were made just for this.
